apps/aesophia/test/contracts/erc20_token.aes

```diff
@@ -0,0 +1,97 @@
+contract ERC20Token =
+  record state = {
+    totalSupply : int,
+    decimals : int,
+    name : string,
+    symbol : string,
+    balances : map(address, int),
+    allowed : map(address, map(address,int)),
+    // Logs, remove when native Events are there
+    transfer_log : list((address,address,int)),
+    approval_log : list((address,address,int))}
+
+  // init(100000000, 10, "Token Name", "TKN")
+  public stateful function init(_totalSupply : int, _decimals : int, _name : string, _symbol : string ) = {
+    totalSupply = _totalSupply,
+    decimals = _decimals,
+    name = _name,
+    symbol = _symbol,
+    balances = {[Call.caller] = _totalSupply }, // creator gets all Tokens
+    allowed = {},
+    // Logs, remove when native Events are there
+    transfer_log = [],
+    approval_log = []}
+
+  public stateful function totalSupply() : int = state.totalSupply
+  public stateful function decimals() : int = state.decimals
+  public stateful function name() : string = state.name
+  public stateful function symbol() : string = state.symbol
+
+  public stateful function balanceOf(tokenOwner : address ) : int = lookupDefault(tokenOwner, state.balances, 0)
+
+  public stateful function transfer(to : address, tokens : int) =
+    put( state{balances[Call.caller] = sub(state.balances[Call.caller], tokens) })
```

Comment:
It seems that the checks here are not trivial, and I see there are some checks in `sub` and `add`. Could you please ensure there is ticket in Pivotal Tracker re updating this contract once `uint` is available in Sophia?

Reply:
Done, PT-160728635